### DISCUSSION TITLE
Does quarkus-mutiny depends on Vert.x under the hood?

### DISCUSSION THREAD

**Discussion by pjk6** [quarkusio/quarkus#30550] [Q&A] [answered]
Hi,

I'm new to Quarkus and i'm currently exploring all reactive libraries and their possibilities.
This article (https://quarkus.io/blog/mutiny-vertx/) describes the relation between Quarkus and Vert.x (quote:  Quarkus is based on Vert.x. Under the hood, there is a managed Vert.x instance that powers the rest of Quarkus). I think that's clear to me.
Quarkus has also an extension 'quarkus-mutiny'. My questions related to this extension are: 

- Does this library also use under the hood Vert.x?
- When this library doesn't depend on Vert.x (it seems that's the case based on the POMs), is it true to say that Quarkus doesn't use Vert.x under the hood in this specific case?
- But suppose the quarkus-mutiny library does use Vert.x under the hood, in which case do we need to use the Smallrye Vert.x Mutiny bindings?

So i'm trying to connect the dots between Quarkus, Vert.x, Mutiny and the Vert.x Mutiny bindings. I hope someone can help me to clarify this a bit. Thanks in advance!

Peter

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment, marked as answer]
Mutiny is a reactive programming library. Vert.x is a reactive stack. They are not related.... but :-D 

To use the reactive model efficiently, you need a way to write asynchronous/non-blocking code: Mutiny. 

See: https://quarkus.io/guides/quarkus-reactive-architecture

**Reply by pjk6** [reply]
Thanks a lot for this clarification! Just for my understanding, I have one question left for you. Suppose that I use Mutiny in a Spring application (anyway an application not based on Vert.x). Is Mutiny in that case working together with Netty (or reactor-netty)?

**Reply by cescoffier** [reply]
First, yes, it's possible.
However, there are no automatic bridge, so you would need to expose Mutiny types on top of reactor netty or netty directly.

**Reply by pjk6** [reply]
Like the SmallRye Mutiny website describes, the Mutiny can be used as a stand-alone library in a Java Project. But are there any examples where Mutiny is used in a Java project which is not based on Vert.x? Because when I google on Mutiny, I always see Mutiny in combination with Vert.x. And I do understand now that Vert.x and Mutiny are unrelated things, but I was just wondering if there are any examples of Mutiny without Vert.x... I'm sorry to bother you again with my questions to better understand the usage of Mutiny. I'll promise its my last question related to this subject :-) Thanks for your quick replies!

**Reply by cescoffier** [reply]
All examples from the mutiny web site are without vert.x
Projects are using Mutiny without Vert.x (typically to improve completion stage-based API), but the main user of Mutiny is Quarkus which runs on top of Vert.x.

**Reply by pjk6** [reply]
Thanks a lot for all your answers. It's becoming more clear now. I appreciate it!
